spec/integration/querying_spec.rb

```diff
@@ -0,0 +1,26 @@
+require "manageiq_helper"
+
+RSpec.describe "Querying" do
+  as_user(:name => "Alice", :userid => "Alice", :password => "alicepassword") do
+    example "queries can be sent in the query string via the GET method" do
+      get(
+        "/graphql",
+        :headers => {"Authorization" => encode_basic_auth_credentials(user.userid, user.password)},
```

Comment:
Note: I don't really want to start propagating Basic Auth tests everywhere, but I think this might be better than using the token auth implementation that we know will change, at least for now.

Reply:
@imtayadeway Sorry, just passing through, but this note intrigued me, so I was curious if this is something that would help with your concern.

What about adding a couple of `get_with_user` and `post_with_user` macros to the existing `spec/support/integration_macros.rb` file, that handle auth based on the existing user?  A quick implementation I whipped up for `get_with_user` (though... I think `post_with_user` would look basically the same):

```ruby
# spec/support/integration_macros.rb
# Note:  This method is an instance method, unlike the `as_user` macro.

  def get_with_user(path, options = {})
    get_options = { :headers => {} }

    case options.delete(:auth)
    when "basic"
      creds = encode_basic_auth_credentials(user.userid, user.password)
      get_options[:headers]["Authorization"] = creds
    when "token"
      # do token version of this...
      get_options[:headers]
    end

    get_options.merge!(options) # might need to be a `deep_merge!` here...

    get(path, get_options)
  end
```

Then the call for this can be shortened to:

```ruby
get_with_user("/graphql", :auth => "basic", :params => {:query => "{ viewer { name } }"})
```

You probably could also include a `:auth_types` keyword to `as_user` macro, that then adds something like a `before { @__auth_type__ = auth_type }` block to the context that can then be used in the case statement like:

```ruby
case options.delete[:auth] || @__auth_type__
when
#...
```

But maybe that is getting a bit too fancy...